frontend/src/components/material/useMaterialViewHelper.js

```diff
@@ -67,6 +67,30 @@ async function getSheets(camp, collection, materialList) {
   )
 }
 
+function randomString(len) {
+  const p = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789'
+  return [...Array(len)].reduce((a) => a + p[~~(Math.random() * p.length)], '')
+}
+
+/**
+ * @param sheets {array} array of {sheetName: string, data: array}
+ * @returns {object} an XLSX workbook
+ */
+export function toWorkbook(sheets) {
+  const workbook = XLSX.utils.book_new()
+  sheets.forEach(({ sheetName, data }) => {
+    const worksheet = XLSX.utils.aoa_to_sheet(data)
+    const validSheetName = sheetName.replaceAll(/[?*[\]/\\:]/g, '')
+    let slicedSheetName = validSheetName.slice(0, 31)
+    if (workbook.SheetNames.includes(slicedSheetName)) {
+      slicedSheetName = validSheetName.slice(0, 28) + randomString(3)
```

Comment:
Why are you using a random string here?
Wouldn't a simple increment be more understandable?

Like:
```
MeinLangerPeriodeNam001
MeinLangerPeriodeNam002
...
```